srtcore/core.cpp

```diff
@@ -329,32 +329,26 @@ void CUDT::setOpt(SRT_SOCKOPT optName, const void* optval, int optlen)
     }
 
     // Post-action, if applicable
-    if (IsSet(oflags, SRTO_POST_SPEC))
+    if (IsSet(oflags, SRTO_POST_SPEC) && m_bConnected)
     {
-        if (m_bConnected)
```

Comment:
Wrong. The `if (m_bConnected)` was intended (together with its braces). This since `if (m_bOpened)` was the merge injection.

Reply:
```c++
if (IsSet(oflags, SRTO_POST_SPEC))
{
    if (m_bConnected)
```
is essentially the same with
```c++
if (IsSet(oflags, SRTO_POST_SPEC) && m_bConnected)
```

Reply:
Ah, missed that, of course.

Actually in the early idea it wasn't a case of every option that it requires the socket to be connected, but it has later turned out that all actions were required for connected sockets only. So, yes, this refax is fair.